src/library_wasmfs.js

```diff
@@ -141,6 +141,28 @@ FS.createPreloadedFile = FS_createPreloadedFile;
       });
     },
     // TODO: read
+    read: (fd, buffer, offset, length, position) => {
+      var seeking = typeof position != 'undefined';
+
+      var dataBuffer = _malloc(length);
+      for (var i = 0; i < length; i++) {
+        {{{ makeSetValue('dataBuffer', 'i', 'buffer[i]', 'i8')}}}
+      }
```

Comment:
I'm confused - this is `read`, so why is it writing the user-provided buffer into memory here?

I would expect to see this allocate `dataBuffer`, call wasmfs to read into there, then copy from there into the output `buffer` the user wants the data in. But I don't think we need to read from `buffer`.